trie.go

```diff
@@ -236,10 +265,7 @@ func collect(node *Node) []string {
 			nodes = append(nodes, c)
 		}
 		if n.term {
-			word := ""
-			for p := n.parent; p.depth != 0; p = p.parent {
-				word = string(p.val) + word
```

Comment:
We might be able to get the same speedup and greatly reduce allocations by using the [string builder](https://golang.org/pkg/strings/#Builder).

Reply:
I did another [PR](Here's another https://github.com/derekparker/trie/pull/22) to optimize collect and now see this one.
In particular, there I do use string builder as you mentioned... might be useful to mix both ideas maybe.